lib/parser.js

```diff
@@ -347,6 +347,12 @@ module.exports = function parser(identifier, string) {
         expect(Token.Type.END_PAREN)
       }
 
+      if (peek().type == Token.Type.DOT) {
+        expect(Token.Type.DOT)
+        token = expect(Token.Type.WORD)
+        name = token.content
```

Comment:
I don't think we want to set this `token.content` to the name.

otherwise `[(foo).bar = 37]` 
becomes `options['bar'] = 37` 
instead of `options['foo'] = { bar: 37 }`